src/composables/transactionList.ts

```diff
@@ -0,0 +1,338 @@
+import { useStore } from 'vuex';
+import { orderBy, uniqBy } from 'lodash-es';
+import {
+  computed,
+  onMounted,
+  ref,
+  watch,
+} from 'vue';
+import { Encoded, Tag } from '@aeternity/aepp-sdk';
+
+import type {
+  ITransaction,
+  ITransactionsState,
+  ITransactionStateList,
+} from '../types';
+import { useMiddleware } from './middleware';
+import {
+  AEX9_TRANSFER_EVENT,
+  fetchJson, getLocalStorageItem, setLocalStorageItem,
+  TX_DIRECTION,
+} from '../popup/utils';
+import { useAccounts } from './accounts';
+import JsonBig from '../lib/json-big';
+import { useAeSdk } from './aeSdk';
+
+const LOCAL_STORAGE_KEY = 'transactions_state';
+
+const isInitalized = ref(false);
+
+const transactions = ref<ITransactionStateList>({});
+
+function generateEmptyTransactionState(): ITransactionsState {
+  return {
+    loaded: [],
+    nextPageUrl: '',
+    pending: {},
+    tipWithdrawnTransactions: [],
+  };
+}
+function getTransactionStateByAccountAddress(address: Encoded.AccountAddress) {
+  return transactions.value[address];
+}
+
+function getOrInitTransactionStateForAccount(address: Encoded.AccountAddress): ITransactionsState {
+  const accountTransactionState = getTransactionStateByAccountAddress(address);
+
+  if (accountTransactionState) {
+    return accountTransactionState;
+  }
+
+  transactions.value[address] = generateEmptyTransactionState();
```

Comment:
I think this is a bad pattern to just add empty data to the state for the reason of returning it. Why not just:
```
return (transactions.value[address])
  ? transactions.value[address]
  : {
    loaded: [],
    nextPageUrl: '',
    pending: {},
    tipWithdrawnTransactions: [],
  };
```